lib/rubocop/cop/rails/exit.rb

```diff
@@ -0,0 +1,40 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop enforces that 'exit' calls are not used within a rails app.
+      # Valid options are instead to raise an error, break, return or some
+      # other form of stopping execution of current request.
+      #
+      # There are two obvious cases where 'exit' is particularly harmful:
+      #
+      # - Usage in library code for your application. Even though rails will
+      # rescue from a SystemExit and continue on, unit testing that library
+      # code will result in specs exiting (potentially silently if exit(0)
+      # is used.)
+      # - Usage in application code outside of the web process could result in
+      # the program exiting, which could result in the code failing to run and
+      # do its job.
+      class Exit < Cop
+        include ConfigurableEnforcedStyle
+
+        def on_send(node)
+          add_offense(node, :selector) if offending_node?(node)
+        end
+
+        private
+
+        def offending_node?(node)
+          _receiver_node, method_name, _arg_nodes = *node
+          method_name == :exit
+        end
+
+        def message(_node)
+          'Favor `fail`, `break`, `return`, etc. over `exit`.'
```

Comment:
This message can simply be a constant. I'd also simplify the message to `Don't use `exit` in Rails applications.` or something like this.